### DISCUSSION TITLE
Connect node-red-contrib-zwave-js to a zwavejs on different server

### DISCUSSION THREAD

**Discussion by maggiagg** [zwave-js/node-red-contrib-zwave-js#257] [Ideas]
I would like the possibility to use this fantastic contrib for different zwave servers. Today, if I understand correctly the zwave stick and the node-red-contrib-zwave-js must run on the same server.

My problem:
- I have 2 different zwave networks at home that I want to control from the same nodered instance
- I have 4 different instances of nodered running on different servers (one for GUI, one for backend, one for the heatpump(critical), etc)
- All of my nodered instances need information from zwavejs - I´m running zwave2mqtt today

Is this an issue that might be of interest for others? Can I help with the implementation in some way.

Thanks for a great job.

**Comment by marcus-j-davies** [comment]
Hi @maggiagg and welcome.

The first thing to note about this module:

It works independently of anything else; it hosts the zwave-js engine directly within Node RED,
it therefore will not be possible to connect this module to an already running zwave-js instance i.e zwavejs2mqtt (Now called ZWave JS UI).

Both cannot be running to control the same network (they both have the same role)

With that said however...


> I have 2 different zwave networks at home that I want to control from the same nodered instance

  This module supports 4 networks (4 x Radios) for each Node RED install.
  Each controller node in your flow is allocated an ID 1-4 (the UI allows you to select what Network ID you want to manage)
  see : https://github.com/zwave-js/node-red-contrib-zwave-js/wiki/User-Interface
  
> I have 4 different instances of nodered running on different servers (one for GUI, one for backend, one for the heatpump(critical), etc)

As I see it, you can have 1 node instance for both of you sticks (max 4 per NR Instance) 

> All of my nodered instances need information from zwavejs - I´m running zwave2mqtt today

This module is not a central Zwave JS Server, but if you did use it, you can quite easily come up with a proprietary protocol so the NR instance hosting your 2 zwave networks can communicate with other NR instances across your network.

Hope that helps
